src/main/scala/com/danielasfregola/zucchini/Dictionary.scala

```diff
@@ -4,16 +4,16 @@ import scala.io.Source
 
 object Dictionary extends App {
 
-
   def extractUniqueWordsFromTextFile(filename: String): Set[String] = {
-    val texts: Array[String] = Source.fromResource(filename).getLines().toArray
+    val source = Source.fromResource(filename)
+    val texts: Iterator[String] = source.getLines()
     val allUniqueWords: Set[String] = allWords(texts).toSet
+    source.close()
     allUniqueWords
   }
 
-  private def allWords(texts: Array[String]): Array[String] = {
-    val words = texts.flatMap(line => line.split(" "))
-    words.map(s => s.filter(_.isLetterOrDigit))
+  private def allWords(texts: Iterator[String]): Iterator[String] = {
+    texts.flatMap(line => line.split("[^a-zA-Z0-9']+")).map(_.trim)
```

Comment:
Much much better! Although probably we can do even better ;)
Have a look at https://stackoverflow.com/a/27311210 and the use of the regex `\W`: this will avoid us from looping through the list of words again to trim

Reply:
@KXA194 No need to do it in this PR, please create a ticket so that we remember to do it as a followup.

Reply:
Will do 😄